File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task81_PERPLEXITY_llama-3-sonar-large-32k-chat.cpp
#include <openssl/x509.h>
#include <openssl/err.h>
#include <openssl/pem.h>
#include <iostream>
#include <string>
#include <iomanip>

int main() {
    const std::string knownHash = "your_known_hash_here";
    const std::string certificatePath = "path_to_your_certificate.cer";

    FILE* file = fopen(certificatePath.c_str(), "r");
    if (!file) {
        std::cerr << "Error opening file" << std::endl;
        return 1;
    }

    X509* cert = PEM_read_X509(file, NULL, NULL, NULL);
    fclose(file);

    if (!cert) {
        std::cerr << "Error reading certificate" << std::endl;
        return 1;
    }

    unsigned char hash[SHA256_DIGEST_LENGTH];
    X509_digest(cert, EVP_sha256(), hash, NULL);

    std::string calculatedHash;
    for (int i = 0; i < SHA256_DIGEST_LENGTH; i++) {
        calculatedHash += std::setw(2) << std::setfill('0') << std::hex << static_cast<int>(hash[i]);
    }

    if (calculatedHash == knownHash) {
        std::cout << "Certificate matches the known hash." << std::endl;
    } else {
        std::cout << "Certificate does not match the known hash." << std::endl;
    }

    X509_free(cert);
    return 0;
}

File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task81_PERPLEXITY_llama-3-sonar-large-32k-chat.java
import java.security.cert.Certificate;
import java.security.cert.CertificateEncodingException;
import java.security.cert.CertificateFactory;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

public class SSLCertificateVerifier {
    public static void main(String[] args) throws CertificateEncodingException, NoSuchAlgorithmException {
        String knownHash = "your_known_hash_here";
        String certificatePath = "path_to_your_certificate.cer";

        CertificateFactory cf = CertificateFactory.getInstance("X.509");
        Certificate cert = cf.generateCertificate(SSLCertificateVerifier.class.getResourceAsStream(certificatePath));

        MessageDigest md = MessageDigest.getInstance("SHA-256");
        byte[] certBytes = cert.getEncoded();
        byte[] hashBytes = md.digest(certBytes);

        StringBuilder sb = new StringBuilder();
        for (byte b : hashBytes) {
            sb.append(String.format("%02x", b));
        }
        String calculatedHash = sb.toString();

        if (calculatedHash.equals(knownHash)) {
            System.out.println("Certificate matches the known hash.");
        } else {
            System.out.println("Certificate does not match the known hash.");
        }
    }
}